### ISSUE TITLE
Binary build?

### ISSUE THREAD

**Issue by gwald** [SonyWWS/LevelEditor#39]
Would be nice to just look at.

**Comment by stoopdapoop** [comment]
@gwald, were you able to get your hands onto one?

I'll be using this project soon and I can add an unmodified release to my fork if it'd still benefit you.

**Comment by gwald** [comment]
Hi, No, at least I dont recall seeing or trying it.
That would be really cool :)

**Comment by stoopdapoop** [comment]
ok, I'll get back to you in a few hours. I'm still at work but I want to check this out once stop working for the day.

**Comment by stoopdapoop** [comment]
@gwald sorry for the delay. Here ye' go.

https://github.com/stoopdapoop/LevelEditor/releases/tag/Release

**Comment by gwald** [comment]
Cheers, thanks :+1: 
By any chance will you be maintaining your fork?
It has high CPU usage for me, I thought it was the rendering (a FPS limiter would be good) but it's still high even without the designview :/

Looks like a cool editor, basic and simple.

**Comment by stoopdapoop** [comment]
I'm not sure, I'm just evaluating this now.

It'll possibly be significant work for me because my engine uses vulkan, so I might need to rewrite the rendering layer, but other than that it seems pretty usable.

I'm not sure what's going on with your cpu usage. Mine is pretty low. I spend most of my cpu time just idling.

What do you want to use this for? I don't imagine it'll be particularly useful if you aren't a programmer.

**Comment by gwald** [comment]
I'm a programmer, just never looked at C#
Not looking to hook in a render or engine, I was looking for an editor that can create a scene file that can be parsed.
The format on is clear XML so it's parse-able.
Ah, I'm using Windows on VMWare, it probably that.

Other editors I've looked at were Horde3D's and Urho3D's editors are very bare bones and Mundus like this just not as advanced.

Thanks again for sharing your build.

**Comment by stoopdapoop** [comment]
Oh my mistake! And yeah, if you've got any other notes from other editors I'd love to see them. I'm looking for a generic editor as well but this is the best I've seen.

If you've got some basic feature requests for this one I'd be happy to hear them. Writing code is more fun when someone actually benefits from your changes.

My biggest concern at the moment is that ATF is a pretty big dependency and it doesn't seem to be getting updated anymore. This may not really be that big of an issue though.

**Comment by gwald** [comment]
Yes, this looks like the best generic editor I've seen also. Generic editors are a thing from history unfortunately.
I remember Mundus, got a fork and was in active development, but this editor is not as advanced as this one IMO.

Horde3d has these options on the editor, sorry image is so bad, Navigation speed & Max FPS:
![image](https://github.com/SonyWWS/LevelEditor/assets/4924613/e377cba0-56e8-4c7e-a410-b2b426916b48)


I think it's very very very important for editors to have a FPS limit (avoiding the game engine/ rendering loop) and it should be very quick and easy to change, ie toolbar FPS field. 5 enter or 100 enter, etc.

The navigation speed is great to change on the fly, quickly also.


An alternative to a game/engine editor is using 3D modellers (that supports scene graphs / nodes) with a custom plugin or command line converter.  Or if you have a basic text based, scene file, a simple text editor or IDE would also work.

**Comment by gwald** [comment]
I started looking at this editor again, and I dont know why, but the renderer keeps looping on idle! but comment out the while loop, fixed the high CPU/GPU usage for me.

https://github.com/SonyWWS/LevelEditor/blob/203a52ac8ab46e8829e5646ac2300cdb0bb68010/LevelEditor/Services/GameLoopService.cs#L47

If anyone wants it... it doesn't count the FPS unless you are doing something, I'm not sure if it brakes anything tho.
[Release.zip](https://github.com/user-attachments/files/16622810/Release.zip)

**Comment by stoopdapoop** [comment]
Hmm, that won't work if your viewport needs to show anything animated.

Also, you can change your update framerate [here,](https://github.com/stoopdapoop/LevelEditor/blob/7f3d04247f8cfbf29ed258dbb5ef66ed62202bfc/LevelEditor/Services/GameLoopService.cs#L182) but it lowering it will not affect your cpu usage much, since this editor "waits" by spinning the CPU as fast as it can until the update time elapses.

@gwald  I'd try something like this instead. I request a higher resolution sleep timer, then I ask to sleep for 
1 millisecond at a time. This makes the idle loop check the elapsed time about 15 times a frame, instead of many thousands of times like it was before. This lowered my cpu usage from 7% to 0%. I suspect you'll see a more dramatic change.
https://github.com/SonyWWS/LevelEditor/compare/master...stoopdapoop:LevelEditor:master

Here's some binaries with the change made.
https://github.com/stoopdapoop/LevelEditor/releases/tag/UnofficialRelease

Let me know if this was useful to you.

**Comment by gwald** [comment]
Excellent! thank you, wow works great :)
Very useful :+1:
